rio_tiler/io/xarray.py

```diff
@@ -72,6 +77,18 @@ def __attrs_post_init__(self):
 
         self.bounds = tuple(self.input.rio.bounds())
         self.crs = self.input.rio.crs
+        if not self.crs:
+            raise MissingCRS(
+                "Dataset doesn't have CRS information, please add it before using rio-tiler (e.g. `ds.rio.write_crs('epsg:4326', inplace=True)`)"
+            )
+
+        if self.crs == WGS84_CRS and (
+            self.bounds[0] < -180
+            or self.bounds[1] < -90
+            or self.bounds[2] > 180
+            or self.bounds[3] > 90
+        ):
```

Comment:
Wondering whether this is robust enough.

Gridded datasets (CDF) are often north-up, but rasters (GeoTIFF etc) often south-up (negative cell/pixel width in y-direction).

I see `self.bounds = tuple(self.input.rio.bounds())` and `rio.bounds` [returns](https://corteva.github.io/rioxarray/html/rioxarray.html#rioxarray.rioxarray.XRasterBase.bounds) `left, bottom, right, top`, possibly taken from an affine transform.

In that case, we could easily encounter datasets in WGS84, but where the y bounds are flipped, which is ok for Rasterio to construct tiles from, I assume, but will fail this check here.

This would make the condition here robust to that case:

```suggestion
        if self.crs == WGS84_CRS and (
            self.bounds[0] < -180
            or min(self.bounds[1], self.bounds[3]) < -90
            or self.bounds[2] > 180
            or max(self.bounds[1], self.bounds[3]) > 90
        ):
```

Reply:
thanks @j08lue 🙏

Reply:
well in fact, rioxarray will `order` the bounds https://github.com/corteva/rioxarray/blob/25fbbce8612ba9a38786a67dc56d24c7d8f79db7/rioxarray/rioxarray.py#L229-L253

Reply:
Ah, they are mixing up "minx/maxx" (ordered projected values) and "left/right" (image orientation) lingo, I think. But the code is clear.